# Análisis Exploratorio de Datos (EDA) sobre Servicios de Internet en Argentina

## Introducción al Análisis Exploratorio de Datos

El acceso a internet se ha convertido en un pilar fundamental para el desarrollo económico y social de Argentina. La conectividad no solo facilita la comunicación y el acceso a la información, sino que también es clave para la educación, la salud, y la economía digital. Sin embargo, la calidad y disponibilidad del servicio de internet varían considerablemente entre regiones, lo que hace necesario un análisis detallado para entender estas diferencias y su evolución en el tiempo.

Este análisis exploratorio de datos (EDA) busca evaluar cómo han cambiado los patrones de uso, la velocidad de conexión, y otros indicadores clave del servicio de internet a nivel nacional entre 2014 y 2024. El objetivo es identificar tendencias y oportunidades para mejorar la infraestructura de telecomunicaciones en todo el país, proporcionando insights que puedan guiar tanto a los responsables de políticas como a las empresas del sector en sus decisiones para mejorar la conectividad en todas las provincias.


-----

## 0. PreAnálisis

Antes de empezar con el análisis mas profundo vi por encima los 6 datasets que elegi para este proyecto, para poder contar que datos registra cada uno:

1. Accesos_por_velocidad.csv: compuesto por casi 985 filas y 11 columnas, nos muestra datos desde 2014 a 2024, divididos por trimestres, teniendo en cuenta las 23 provincias argentinas y la capital federal, el rango de banda ancha y cuantos hogares tienen acceso en esas localidades a cada rango de velocidad.

2. Totales_Accesos_por_velocidad.csv: compuesto por 42 filas y 12 columnas, muestra datos desde 2014 a 2024, de distintos rangos de banda ancha(velocidad), y cuantos hogares(es lo que supongo por los numeros) tienen acceso a esos rangos de banda ancha.

3. Velocidad_por_provincia.csv: compuesto por 985 filas y 4 columnas, muestra datos desde 2014 a 2024, divididos por trimestres, de las 23 provincias argentinas y la capital federal, cual es la media de bajada de cada una en Mbps.

4. Internet_IngresosTotalesxTrimestre.csv: compuesto por 43 filas y 4 columnas, muestra datos desde 2014 a 2024, divididos por trimestre, de la cantidad ingresos totales de los servicios de Internet a nivel nacional.

5. Penetracion-hogares.csv: compuesto por 985 filas y 4 columnas, muestra datos desde 2014 a 2024, divididos por trimestre, mostrando los accesos cada 100 hogares por provincia.

6. Penetracion-totales.csv: compuesto por 42 filas y 5 columnas, muestrados desde 2014 a 2024, divididos por trimestre, mostrando los accesos cada 100 hogares, y tambien por cada 100 habitantes, en total del pais.


Empiezo importando las librerias necesarias para realizar el análisis

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np


Elijo los CSV que me parecen importantes para realizarles un análisis, teniendo en cuenta que necesito conocer los aspectos mas importantes del comportamiento de los servicios de Internet a nivel nacional (Argentina).

In [ ]:
# Leo y cargo en un DataFrame cada uno de los 6 CSVs que elegi guiandome por los datos que vi por encima en libreOffice
accesos_Provincia_Por_Velocidad = pd.read_csv('Dataset_Internet/00Accesos/Accesos_por_velocidad.csv', encoding='latin1')
totalesarg_Accesos_Por_Velocidad = pd.read_csv('Dataset_Internet/00Accesos/Totales_Accesos_por_velocidad.csv', encoding='latin1')
velocidad_Total_Por_Provincia = pd.read_csv('Dataset_Internet/03Velocidad/Velocidad_por_provincia.csv', encoding='latin1')
internet_IngresosTotalesxTrimestre = pd.read_csv('Dataset_Internet/02Totales/Internet_IngresosTotalesxTrimestre.csv', encoding='latin1')
penetracion_Hogares = pd.read_csv('Dataset_Internet/01Penetracion/Penetracion-hogares.csv', encoding='latin1')
penetracion_Totalesarg = pd.read_csv('Dataset_Internet/01Penetracion/Penetracion-totales.csv', encoding='latin1')


## Mi enfoque en el análisis

Ahora voy a ir por cada uno de los datasets analizando detenidamente las siguientes cosas:
1. Aspectos Generales del dataset: 
    Como el tipo de variables (ya sean categoricas o numericas), la cantidad de filas y columnas, valores nulos o faltantes.
2. Descripcion Estadistica:
 
    Medidas de tendencia central:
    
    Media
    
    Mediana

    Medidas de Variabilidad:

    Desviacion Estandar
    
    Rango Intercuartil

3. Visualizacion de los datos:
    Histograma
    Boxplot (para intercuartiles)

4. Analisis Bivariado y Multivariado:
    Grafica de dispersion >>> Para ver aumento lineal
    Indice de Correlacion Lineal entre variables: Positiva, Inversa o no hay Correlacion.
    Grafico de Violin
    Comparar una variable categorica y una numerica
5. Correlacion

----

1. ## Análisis de accesos_Provincia_Por_Velocidad

In [ ]:
accesos_Provincia_Por_Velocidad.head()

Con este .head() miro por encima que tengo datos categoricos, y datos numericos discretos. Ademas de la distribucion de las columnas.

In [ ]:
accesos_Provincia_Por_Velocidad.info()

De este .info() puedo ver los tipos de datos de cada columnas, donde tenemos integers, floats y objetos, determino tambien que solo hay valores faltantes en la columna OTROS, luego de revisarlos con una herramienta llamada "Data Wrangler" que me permite ver el dataset completo, llegue a la conclusion de que puedo transformar los nulos a 0 para tener mas limpio el df.

In [ ]:
accesos_Provincia_Por_Velocidadsinnulos = accesos_Provincia_Por_Velocidad.fillna(0)

In [ ]:
accesos_Provincia_Por_Velocidadsinnulos.head()

Con el siguiente .describe() puedo ver datos como la media, la variacion estandar, los valores minimos y maximos y los cuartiles.

In [ ]:
accesos_Provincia_Por_Velocidadsinnulos.describe()

In [133]:
accesos_Provincia_Por_Velocidadsinnulos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  984 non-null    int64  
 1   Trimestre            984 non-null    int64  
 2   Provincia            984 non-null    object 
 3   HASTA 512 kbps       984 non-null    float64
 4   + 512 Kbps - 1 Mbps  984 non-null    float64
 5   + 1 Mbps - 6 Mbps    984 non-null    object 
 6   + 6 Mbps - 10 Mbps   984 non-null    float64
 7   + 10 Mbps - 20 Mbps  984 non-null    float64
 8   + 20 Mbps - 30 Mbps  984 non-null    float64
 9   + 30 Mbps            984 non-null    object 
 10  OTROS                984 non-null    float64
 11  Total                984 non-null    object 
dtypes: float64(6), int64(2), object(4)
memory usage: 92.4+ KB


Paso el csv limpio de accesos_Provincia_Por_Velocidad a un archivo .csv

In [ ]:
accesos_Provincia_Por_Velocidadsinnulos.to_csv('accesos_Provincia_Por_Velocidad_Limpio.csv', index=False)

----

## 2. Análisis de totalesarg_Accesos_Por_Velocidad

Repito el mismo proceso hecho anteriormente ahora con el siguiente dataframe

In [ ]:
totalesarg_Accesos_Por_Velocidad.head()

Con el .info() me doy cuenta de que todas las filas tienen datos que no son nulos, por lo cual decido no transformar este dataset.

In [132]:
totalesarg_Accesos_Por_Velocidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Año                      41 non-null     int64  
 1   Trimestre                41 non-null     int64  
 2   Hasta 512 kbps           41 non-null     float64
 3   Entre 512 Kbps y 1 Mbps  41 non-null     float64
 4   Entre 1 Mbps y 6 Mbps    41 non-null     object 
 5   Entre 6 Mbps y 10 Mbps   41 non-null     object 
 6   Entre 10 Mbps y 20 Mbps  41 non-null     object 
 7   Entre 20 Mbps y 30 Mbps  41 non-null     object 
 8   Más de 30 Mbps           41 non-null     object 
 9   OTROS                    41 non-null     float64
 10  Total                    41 non-null     object 
dtypes: float64(3), int64(2), object(6)
memory usage: 3.7+ KB


In [ ]:
totalesarg_Accesos_Por_Velocidad.describe()

----

## 3. Análisis de velocidad_Total_Por_Provincia

Reviso de que trata el dataframe, veo que tengo datos categoricos y numericos

In [ ]:
velocidad_Total_Por_Provincia.head()

Veo que no hay nulos asi que no transformo los datos, se los tipos de datos que tengo, tanto objetos como integers, la cantidad de filas(984) y la cantidad de columnas(4)

In [131]:
velocidad_Total_Por_Provincia.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Año                     984 non-null    int64 
 1   Trimestre               984 non-null    int64 
 2   Provincia               984 non-null    object
 3   Mbps (Media de bajada)  984 non-null    object
dtypes: int64(2), object(2)
memory usage: 30.9+ KB


In [ ]:
velocidad_Total_Por_Provincia.describe()

---

## 4. Análisis de internet_IngresosTotalesxTrimestre

In [ ]:
internet_IngresosTotalesxTrimestre.head()

In [130]:
internet_IngresosTotalesxTrimestre.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Año                        41 non-null     int64 
 1    Trimestre                 41 non-null     int64 
 2   Ingresos (miles de pesos)  41 non-null     object
 3    Periodo                   41 non-null     object
dtypes: int64(2), object(2)
memory usage: 1.4+ KB


In [ ]:
internet_IngresosTotalesxTrimestre.describe()

---

## 5. Análisis de penetracion_Hogares

In [ ]:
penetracion_Hogares.head()

In [129]:
penetracion_Hogares.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           984 non-null    int64 
 1    Trimestre                    984 non-null    int64 
 2    Provincia                    984 non-null    object
 3   Accesos por cada 100 hogares  984 non-null    object
dtypes: int64(2), object(2)
memory usage: 30.9+ KB


In [ ]:
penetracion_Hogares.describe()

----

## 6. Análisis de penetracion_Totalesarg

In [ ]:
penetracion_Totalesarg.head()

In [128]:
penetracion_Totalesarg.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           41 non-null     int64 
 1   Trimestre                     41 non-null     int64 
 2   Accesos por cada 100 hogares  41 non-null     object
 3   Accesos por cada 100 hab      41 non-null     object
 4   Periodo                       41 non-null     object
dtypes: int64(2), object(3)
memory usage: 1.7+ KB


In [ ]:
penetracion_Totalesarg.describe()

----